### ISSUE TITLE
[Critical] TF-Diffusion-Training mit Live-Encoder bricht immer ab: 'float' object has no attribute 'detach'

### ISSUE THREAD

**Issue by Raphaleliomm** [Raphaleliomm/Minecraft-AI-Structure-Generator#1]
## Problem

`train_transformer_diffusion_step()` ruft `float(loss.detach())` auf, obwohl `chunked_weighted_ce_loss()` ein Python-`float` (keinen Tensor) zurueckgibt -> `AttributeError` schon im ersten Mikro-Batch.

* `app/diffusion_model.py:778` und `app/diffusion_model.py:780` -> `return float(loss.detach())`
* `app/diffusion_model.py:672-695` -> `chunked_weighted_ce_loss()` summiert mit `float(chunk_loss.detach())` und `return total_loss`, liefert also einen `float`
* Nur dieser Pfad ist betroffen: `train_diffusion_step` (`:727`) und `train_transformer_diffusion_step_cached` (`:826/:828`) machen es korrekt ueber `float(loss)`.

## Reproduktion (CPU, Mini-Modell, Grid 8^3)

```
chunked_weighted_ce_loss -> float 1.5671499222517014
train_transformer_diffusion_step (live encoder) FAILED -> AttributeError 'float' object has no attribute 'detach'
train_transformer_diffusion_step_cached OK -> 1.96486496925354
```

## Auswirkung

* `app/main_app.py:2077`: Der Zweig **TF-Diffusion ohne Hidden-State-Cache (Live-Encoder)** kann nie trainieren. Die Exception wird bei `app/main_app.py:2125` nur abgefangen und als `tf_error` in die Loss-Zeile geschrieben, der Lauf endet nach 0 Epochen.
* `kaggle_export.py:558`: Der erzeugte Kaggle-Notebook-Code ruft genau diese Funktion auf -> auch der Export bricht im ersten Batch ab.
* Die Tests decken nur den gecachten Pfad ab (`tests/test_grad_accum.py:295`, `tests/test_train_more.py`), deshalb ist es nicht aufgefallen.

## Fix-Vorschlag

1. In `train_transformer_diffusion_step` konsequent `float(loss)` verwenden (analog zur gecachten Variante).
2. Regressionstest ergaenzen, der `train_transformer_diffusion_step` mit einem Fake-Encoder auf einem Mini-Modell ausfuehrt (Muster von `tests/test_grad_accum.py`).
